momentGW/thc.py

```diff
@@ -0,0 +1,270 @@
+import numpy as np
+from h5py import File
+from pyscf import lib
+from pyscf.agf2 import mpi_helper
+from scipy.special import binom
+
+
+class Integrals:
+    """
+    Container for the integrals required for GW methods.
+    """
+
+    def __init__(
+        self,
+        with_df,
+        mo_coeff,
+        mo_occ,
+        thc_opts,
+    ):
+        self.with_df = with_df
+        self.mo_coeff = mo_coeff
+        self.mo_occ = mo_occ
+        self.filepath = thc_opts["file_path"]
+
+        self._blocks = {}
+
+    def transform(self):
+        """
+        Imports a H5PY file containing a dictionary. Inside the dict, a 'collocation_matrix' and
+        a 'coulomb_matrix' must be contained with shapes (aux, MO) and (aux,aux) respectively.
+        """
+        if self.filepath is None:
+            raise ValueError("filepath cannot be None for THC implementation")
+        thc_eri = File(self.filepath, "r")
+        coll = np.array(thc_eri["collocation_matrix"]).T[0].T
+        cou = np.array(thc_eri["coulomb_matrix"][0]).T[0].T
+        Xip = coll[: self.nocc, :]
+        Xap = coll[self.nocc :, :]
+        self._blocks["coll"] = coll
+        self._blocks["cou"] = cou
+        self._blocks["Xip"] = Xip
+        self._blocks["Xap"] = Xap
+
+    @property
+    def Coll(self):
+        """
+        Return the (aux, MO) array.
+        """
+        return self._blocks["coll"]
+
+    @property
+    def Cou(self):
+        """
+        Return the (aux, aux) array.
+        """
+        return self._blocks["cou"]
+
+    @property
+    def Xip(self):
+        """
+        Return the (aux, W occ) array.
+        """
+        return self._blocks["Xip"]
+
+    @property
+    def Xap(self):
+        """
+        Return the (aux, W vir) array.
+        """
+        return self._blocks["Xap"]
+
+    @property
+    def nmo(self):
+        """
+        Return the number of MOs.
+        """
+        return self.mo_coeff.shape[-1]
+
+    @property
+    def nocc(self):
+        """
+        Return the number of occupied MOs.
+        """
+        return np.sum(self.mo_occ > 0)
+
+    @property
+    def naux(self):
+        """
+        Return the number of auxiliary basis functions, after the
+        compression.
+        """
+        return self.Cou.shape[0]
+
+
+class TDA:
```

Comment:
This should inherit the other `TDA`, i.e. add at the top
`from momentGW import tda`
and then change this line to 
`class TDA(tda.TDA):`